mountpoint-s3/src/fs.rs

```diff
@@ -726,10 +698,29 @@ where
             return Err(err!(libc::EINVAL, "O_SYNC and O_DSYNC are not supported"));
         }
 
-        // All file handles will be lazy initialized on first read/write.
-        let state = FileHandleState::new(lookup, flags, pid).await.into();
+        let state = if flags & libc::O_RDWR != 0 {
```

Comment:
This is worth a change log entry, you're right.

We made changes to tests to accommodate this change.

Reply:
Yes, most of the changes in the tests are due to this change. Note that this is reverting a change made when introducing overwrite support, which was not explicitly documented in the first place.

Reply:
I see, so there is no place in existing documentation which we could reference and say "this is changing now, errors will be reported earlier". But probably worth bridging this gap in new CHANGELOG.md entry? Something like:
> Release v1.4.0 made some errors reported on first access instead of `open()`, changes in v1.4.1 reverts this to reporting on `open()`

Reply:
I've added a note to the change log in the next push.